src/Dictionaries/RegExpTreeDictionary.cpp

```diff
@@ -867,12 +867,17 @@ void registerDictionaryRegExpTree(DictionaryFactory & factory)
         String dictionary_layout_prefix = config_prefix + ".layout" + ".regexp_tree";
         const DictionaryLifetime dict_lifetime{config, config_prefix + ".lifetime"};
 
-        RegExpTreeDictionary::Configuration configuration{
-            .require_nonempty = config.getBool(config_prefix + ".require_nonempty", false), .lifetime = dict_lifetime};
-
         const auto dict_id = StorageID::fromDictionaryConfig(config, config_prefix);
 
         auto context = copyContextAndApplySettingsFromDictionaryConfig(global_context, config, config_prefix);
+        const auto * clickhouse_source = typeid_cast<const ClickHouseDictionarySource *>(source_ptr.get());
```

Comment:
why here it is `typeid_cast`?

Reply:
As far as `class ClickHouseDictionarySource final` it shouldn't be any difference between `typeid_cast` and `dynamic_cast`